File: ruby/primary_account/remove_user.rb
#!/usr/bin/env ruby
# Encoding: utf-8
#
# Copyright:: Copyright 2016, Google Inc. All Rights Reserved.
#
# License:: Licensed under the Apache License, Version 2.0 (the "License");
#           you may not use this file except in compliance with the License.
#           You may obtain a copy of the License at
#
#           http://www.apache.org/licenses/LICENSE-2.0
#
#           Unless required by applicable law or agreed to in writing, software
#           distributed under the License is distributed on an "AS IS" BASIS,
#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
#           implied.
#           See the License for the specific language governing permissions and
#           limitations under the License.
#
# Removes a user from the primary account.

require_relative '../shopping_common'

def remove_user(content_api, merchant_id, email_address)
  # First we need to retrieve the full object, so we know the existing set of
  # users.
  # No point retrieving the entire object when we just want the list of users.
  account = content_api.get_account(
      merchant_id, merchant_id, fields: 'users'
  ) do |res, err|
    if err
      if err.status == 404
      puts "Account #{merchant_id} not found."
    else
      handle_errors(err)
      end
      exit
    end
  end

  # Remove user from user list.
  account.users.reject! { |user| user.email_address == email_address }

  # Patch account with new user list.
  response = content_api.patch_account(merchant_id, merchant_id, account) do
      |res, err|
    if err
      handle_errors(err)
      exit
    end

    puts "User #{email_address} removed from account #{merchant_id}."
  end
end


if __FILE__ == $0
  options = ArgParser.parse(ARGV)
  config, content_api = service_setup(options)
  email_address = config.account_sample_user
  if email_address.nil? or email_address.empty?
    puts "No account sample user address in the configuration."
    exit
  end
  remove_user(content_api, config.merchant_id, email_address)
end

File: python/shopping/content/accounts/remove_user.py
#!/usr/bin/python
#
# Copyright 2016 Google Inc. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""Removes a user from the primary account."""

from __future__ import print_function
import sys

from shopping.content import common


def main(argv):
  # Authenticate and construct service.
  service, config, _ = common.init(argv, __doc__)
  merchant_id = config['merchantId']
  email = config.get('accountSampleUser')
  if not email:
    print('Must specify the user email to remove in the samples configuration.')
    sys.exit(1)

  # First we need to retrieve the existing set of users.
  account = service.accounts().get(
      merchantId=merchant_id, accountId=merchant_id).execute()

  users = account.get('users')
  if not users:
    print('No users in account %d.' % merchant_id)
    sys.exit(1)

  matched = [u for u in users if u['emailAddress'] == email]
  if not matched:
    print('User %s was not found.' % email)
    sys.exit(1)

  for u in matched:
    users.remove(u)
  account['users'] = users

  # Patch account with new user list.
  service.accounts().update(
      merchantId=merchant_id, accountId=merchant_id, body=account).execute()

  print('User %s was removed from merchant ID %d' % (email, merchant_id))


if __name__ == '__main__':
  main(sys.argv)
